### DISCUSSION TITLE
Select users with their roles name?

### DISCUSSION THREAD

**Discussion by shinoaq** [spatie/laravel-permission#2060] [Q&A]
I want to select user list with only just role name without any extra info about role, how i can do it?
When i use `User::with('roles')->get();` i get extra info like: guard_name, created_at, pivot and etc. i Just need role name

**Comment by erikn69** [comment]
Use `map` on collections

**Reply by shinoaq** [reply]
can u show any example pls?

**Reply by erikn69** [reply]
https://laravel.com/docs/9.x/collections#method-map

Something like this
```php
User::with('roles')->get()->map(function ($user) {
    $user->roles = $user->roles->map(fn($role) => $role->name);
    $user->unsetRelation('roles');
    return $user;
});
```
